src/pip/_internal/operations/install/wheel.py

```diff
@@ -135,24 +136,23 @@ def message_about_scripts_not_on_PATH(scripts: Sequence[str]) -> Optional[str]:
         return None
 
     # Group scripts by the path they were installed in
-    grouped_by_dir: Dict[str, Set[str]] = collections.defaultdict(set)
+    grouped_by_dir: Dict[Path, Set[str]] = collections.defaultdict(set)
     for destfile in scripts:
-        parent_dir = os.path.dirname(destfile)
-        script_name = os.path.basename(destfile)
+        parent_dir = Path(destfile).parent.resolve()
+        script_name = Path(destfile).name
```

Comment:
```suggestion
        dest_path = Path(destfile)
        parent_dir = dest_path.parent.resolve()
        script_name = dest_path.name
```

Avoid creating the object repeatedly